### PR TITLE
MsGraphicsPkg: More CodeQL fixes

### PR DESCRIPTION
**Pull request by TaylorBeebe**
## Description

Various fixes

- [x] Impacts functionality?
- **Functionality** - Does the change ultimately impact how firmware
functions?
- Examples: Add a new library, publish a new PPI, update an algorithm,
...
- [x] Impacts security?
- **Security** - Does the change have a direct security impact on an
application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
- **Breaking change** - Will anyone consuming this change experience a
break
    in build or boot behavior?
- Examples: Add a new library class, move a module to a different repo,
call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
- **Documentation** - Does the change contain explicit documentation
additions
    outside direct code modifications (and comments)?
- Examples: Update readme file, add feature readme file, link to
documentation
    on an a separate Web page, ...

## How This Was Tested

Build and boot changes on QemuQ35Pkg to EFI shell.

## Integration Instructions

N/A

### CODE REVIEW

File: MsGraphicsPkg/SimpleWindowManagerDxe/WaitForEvent.c [hunk 1]
```diff
@@ -153,14 +153,21 @@ WaitForEventInternal (
...
-                           PowerTitle,                               // Dialog body title.
-                           PowerBody,                                // Dialog body text.
-                           PowerCaption,                             // Dialog caption text.
-                           SWM_MB_CANCEL | SWM_MB_STYLE_ALERT2,      // Show Cancel and enforce Timeout
-                           EFI_TIMER_PERIOD_SECONDS (PcdGet16 (PcdPowerOffHold)),
-                           &SwmResult
-                           );                                         // Return result.
+          // MU_CHANGE [BEGIN] - CodeQL change
+          if ((PowerTitle == NULL) || (PowerBody == NULL) || (PowerCaption == NULL)) {
+            DEBUG ((DEBUG_ERROR, "Error getting POWER OFF dialog strings. Code = %r\n", Status));
```
Comment (kenlautner) [inline MsGraphicsPkg/SimpleWindowManagerDxe/WaitForEvent.c]: Shouldn't you be updating the status here?  It should always be EFI_SUCCESS at this point.

**Review by kenlautner — APPROVED** [review]

**Review by makubacki — APPROVED** [review]

File: MsGraphicsPkg/DisplayEngineDxe/FormDisplay.c [hunk 2]
```diff
@@ -636,8 +641,13 @@ UiAddMenuOption (
 
   for (Index = 0; Index < Count; Index++) {
     MenuOption = AllocateZeroPool (sizeof (UI_MENU_OPTION));
-    ASSERT (MenuOption);
+    // MU_CHANGE [BEGIN] - CodeQL change
+    if (MenuOption == NULL) {
+      ASSERT (MenuOption);
```
Comment (os-d) [inline MsGraphicsPkg/DisplayEngineDxe/FormDisplay.c]: nit: since we are changing it anyway, change to usual style of `MenuOption != NULL`?

**Review by os-d — APPROVED** [review]

File: MsGraphicsPkg/DisplayEngineDxe/FormDisplay.c (hunk 2 above)
Comment (TaylorBeebe) [reply MsGraphicsPkg/DisplayEngineDxe/FormDisplay.c]: Updated

File: MsGraphicsPkg/SimpleWindowManagerDxe/WaitForEvent.c (hunk 1 above)
Comment (TaylorBeebe) [reply MsGraphicsPkg/SimpleWindowManagerDxe/WaitForEvent.c]: Updated

**Pull request merged**